cocos/core/components/renderer.ts

```diff
@@ -35,31 +35,16 @@ import {
 import { Material } from '../assets/material';
 import { Component } from './component';
 import { IMaterialInstanceInfo, MaterialInstance } from '../renderer/core/material-instance';
-import { scene } from '../renderer';
-import { Layers } from '../scene-graph/layers';
 import { warnID } from '../platform/debug';
-import { legacyCC } from '../global-exports';
 
 const _matInsInfo: IMaterialInstanceInfo = {
     parent: null!,
     owner: null!,
     subModelIdx: 0,
 };
 
-@ccclass('cc.RenderableComponent')
-export class RenderableComponent extends Component {
-    @serializable
-    protected _visFlags = Layers.Enum.NONE;
-
-    get visibility () {
-        return this._visFlags;
-    }
-
-    set visibility (val) {
-        this._visFlags = val;
-        this._onVisibilityChange(val);
-    }
-
+@ccclass('cc.Renderer')
+export class Renderer extends Component {
```

Comment:
Renderer is still too general for me, maybe AbstractRenderer, would that sound better @minggo